### ISSUE TITLE
Display backend version in footer

### ISSUE THREAD

**Issue by ozamosi** [weaveworks/weave-gitops#2117]
When receiving a bug report, we want to know the exact version that the bug report is against. For releases, this is currently shown in the footer. For development versions, we show the last release version as described in the javascript. This means, if there's a long gap with lots of fixes between 0.8 and 0.9, we won't be able to easily spot if a problem report is newer or older than the fix that was supposed to fix it.

In #1953 we added a new backend endpoint that reports the exact build version.

Add a hook that calls the version endpoint using the generated API client, and use that hook to print the backend version in the footer component instead.

**Comment by JamWils** [comment]
```
{"version":{"branch":"", "buildtime":"", "git-commit":"", "version":"v0.0.0"}}
```

If the version from the endpoint matches the JSON version then print the version number. E.g, `v0.8.1`

If they do not match then default to branch and git commit `staging-123cbde4`

@SamLR does this make sense?

**Comment by SamLR** [comment]
@JamWils when you say 'JSON version' do you mean the version as set in package.json (that's available to the UI)? Assuming that, yes I think that makes sense.

**Comment by JamWils** [comment]
That is correct, that is what I meant.  Thanks for the clarification.  @opudrovs let me know if I can provide any additional clarification on the above.

**Comment by opudrovs** [comment]
@JamWils previously displayed version (from `package.json`) is linked to

`https://github.com/weaveworks/weave-gitops/releases/tag/v${p.version}`

What should the link point to if the branch and commit are displayed (`staging-123cbde4`)? To the commit via short commit hash (`version["git-commit"]`), like this?

`https://github.com/weaveworks/weave-gitops/commit/123cbde4`

**Comment by opudrovs** [comment]
@JamWils  this is what the data returned looks like in staging:

`{"version":{"branch":"main", "buildtime":"2022-05-16_10:56:21", "git-commit":"7aa118f", "version":"7aa118f"}}`
